src/utils/flow.js

```diff
@@ -1,16 +1,25 @@
 /* @flow */
 import type { FlowType, FlowTypes } from "./types";
 
-function flowTypeAnnotationToString(type: TypeTypeAnnotation, nullable: boolean): FlowType {
+function flowTypeAnnotationToString(type: TypeTypeAnnotation, nullable: boolean, variables?: Object): FlowType {
   switch (type.type) {
     case "BooleanTypeAnnotation":
-      return { type: "boolean", nullable };
+      return { type: "boolean", nullable, variables };
     case "NumberTypeAnnotation":
-      return { type: "number", nullable };
+      return { type: "number", nullable, variables };
     case "StringTypeAnnotation":
-      return { type: "string", nullable };
+      return { type: "string", nullable, variables };
+    case "StringLiteralTypeAnnotation":
+    case "BooleanLiteralTypeAnnotation":
+    case "NumericLiteralTypeAnnotation":
```

Comment:
Maybe it's worth following GraphQL and having Input and Output types? A literal type would never be valid as an Output.